incubator/hnc/internal/validators/hierarchy.go

```diff
@@ -210,9 +212,55 @@ func (v *Hierarchy) checkParent(ns, curParent, newParent *forest.Namespace) admi
 		return deny(metav1.StatusReasonConflict, "Illegal parent: "+reason)
 	}
 
+	// Prevent overwriting source objects in the descendants after the hierarchy change.
+	if yes, co := v.hasConflictObject(newParent, ns); yes {
+		coStr := strings.Join(co, "\n  * ")
+		msg := fmt.Sprintf("\nCannot update hierarchy because it would overwrite the following object(s):\n  * %s\nTo fix this, please rename or remove the conflicting objects first.", coStr)
```

Comment:
I'm not sure about that _leading_ newline, I've never seen that before so it's probably a good idea to have at least the first part of the message in the main line.

Nit: also, it might be more clear to construct the message like this:

```
msg := "Cannot update hierarchy because it would overwrite the following object(s):\n"
msg += "  * " + coStr + "\n"
msg += "To fix this, please rename or remove the conflicting objects first."
```